src/coreclr/src/vm/dllimport.cpp

```diff
@@ -4179,15 +4191,21 @@ static void CreateNDirectStubAccessMetadata(
     }
 
     int lcidArg = -1;
+
+    // Check if we have a MethodDesc to query for additional data.
     if (pSigDesc->m_pMD != NULL)
     {
-        lcidArg = GetLCIDParameterIndex(pSigDesc->m_pMD);
+        MethodDesc* pMD = pSigDesc->m_pMD;
+
+        // P/Invoke marked with UnmanagedCallersOnlyAttribute is not
+        // presently supported.
+        if (pMD->HasUnmanagedCallersOnlyAttribute())
+            COMPlusThrow(kNotSupportedException, IDS_EE_NDIRECT_UNSUPPORTED_SIG);
```

Comment:
Could you please also add the same check to `PInvokeILEmitter.EmitIL` in `src\coreclr\src\tools\aot\ILCompiler.ReadyToRun\IL\Stubs\PInvokeILEmitter.cs` ?